src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs

```diff
@@ -17,25 +17,14 @@ namespace Microsoft.Azure.WebJobs.Host.Executors
 {
     internal class DefaultStorageCredentialsValidator : IStorageCredentialsValidator
     {
-        private readonly HashSet<StorageCredentials> _validatedCredentials = new HashSet<StorageCredentials>();
-
         public async Task ValidateCredentialsAsync(IStorageAccount account, CancellationToken cancellationToken)
         {
             if (account == null)
             {
                 throw new ArgumentNullException("account");
             }
 
-            StorageCredentials credentials = account.Credentials;
-
-            // Avoid double-validating the same account and credentials.
-            if (_validatedCredentials.Contains(credentials))
-            {
-                return;
-            }
-
             await ValidateCredentialsAsyncCore(account, cancellationToken);
```

Comment:
now that you're no longer performing additional logic in this method, it seems we can get rid of the "Core" method and move the logic back into this method?